proofs/sectorbuilder/rustsectorbuilder.go

```diff
@@ -192,104 +138,83 @@ func (sb *RustSectorBuilder) AddPiece(ctx context.Context, pieceRef cid.Cid, pie
-			cPieceKey,
-			C.uint64_t(pieceSize),
-			cSinkPath,
-		)))
-		defer C.sector_builder_ffi_destroy_add_piece_response(resPtr)
-
-		if resPtr.status_code != 0 {
-			msg := "CGO add_piece returned an error (error_msg=%s, sinkPath=%s)"
-			log.Errorf(msg, C.GoString(resPtr.error_msg), sink.ID())
-			errCh <- errors.New(C.GoString(resPtr.error_msg))
+		id, err := libsectorbuilder.AddPiece(sb.ptr, pieceRef.String(), pieceSize, fifoFile.ID())
+		if err != nil {
+			msg := "CGO add_piece returned an error (err=%s, fifo path=%s)"
+			log.Errorf(msg, err, fifoFile.ID())
+			errCh <- err
 			return
 		}
 
-		sectorIDCh <- uint64(resPtr.sector_id)
+		sectorIDCh <- id
 	}()
 
 	select {
 	case <-ctx.Done():
 		errStr := "context completed before CGO call could return"
 		strFmt := "%s (sinkPath=%s)"
-		log.Errorf(strFmt, errStr, sink.ID())
+		log.Errorf(strFmt, errStr, fifoFile.ID())
 
 		return 0, errors.New(errStr)
 	case err := <-errCh:
 		errStr := "error streaming piece-bytes"
 		strFmt := "%s (sinkPath=%s)"
-		log.Errorf(strFmt, errStr, sink.ID())
+		log.Errorf(strFmt, errStr, fifoFile.ID())
 
 		return 0, errors.Wrap(err, errStr)
 	case sectorID := <-sectorIDCh:
 		go sb.sealStatusPoller.addSectorID(sectorID)
-		log.Infof("add piece complete (pieceRef=%s, sectorID=%d, sinkPath=%s)", pieceRef.String(), sectorID, sink.ID())
+		log.Infof("add piece complete (pieceRef=%s, sectorID=%d, sinkPath=%s)", pieceRef.String(), sectorID, fifoFile.ID())
 
 		return sectorID, nil
 	}
 }
 
 func (sb *RustSectorBuilder) findSealedSectorMetadata(sectorID uint64) (*SealedSectorMetadata, error) {
-	resPtr := (*C.sector_builder_ffi_GetSealStatusResponse)(unsafe.Pointer(C.sector_builder_ffi_get_seal_status((*C.sector_builder_ffi_SectorBuilder)(sb.ptr), C.uint64_t(sectorID))))
-	defer C.sector_builder_ffi_destroy_get_seal_status_response(resPtr)
-
-	if resPtr.status_code != 0 {
-		return nil, errors.New(C.GoString(resPtr.error_msg))
+	status, err := libsectorbuilder.GetSectorSealingStatusByID(sb.ptr, sectorID)
+	if err != nil {
+		return nil, err
 	}
 
-	if resPtr.seal_status_code == C.Failed {
-		return nil, errors.New(C.GoString(resPtr.seal_error_msg))
-	} else if resPtr.seal_status_code == C.Pending {
-		return nil, nil
-	} else if resPtr.seal_status_code == C.Sealing {
-		return nil, nil
-	} else if resPtr.seal_status_code == C.Sealed {
-		commRSlice := goBytes(&resPtr.comm_r[0], 32)
-		var commR types.CommR
-		copy(commR[:], commRSlice)
-
-		commDSlice := goBytes(&resPtr.comm_d[0], 32)
-		var commD types.CommD
-		copy(commD[:], commDSlice)
-
-		commRStarSlice := goBytes(&resPtr.comm_r_star[0], 32)
-		var commRStar types.CommRStar
-		copy(commRStar[:], commRStarSlice)
-
-		proof := goBytes(resPtr.proof_ptr, resPtr.proof_len)
-
-		ps, err := goPieceInfos(resPtr.pieces_ptr, resPtr.pieces_len)
-		if err != nil {
-			return nil, errors.Wrap(err, "failed to marshal from string to cid")
-		}
-
+	if status.SealStatusCode == 0 {
```

Comment:
I'd prefer it if these numbers were descriptive constants, but at least the comment is there, which serves the same purpose.

Reply:
Cool. I’ll factor these out into some enum-like thingy ASAP.